js/script.js

```diff
@@ -1,267 +1,79 @@
-                cards.removeChild(event.path[2]); //удаляем
-            }else{//вся карточка за исключением иконок лайк и удаления
-                if (event.target.classList.contains("place-card__image")){//картинка, а не подписи внизу
-                    const  popupImage= document.querySelector('.popup__image');
-                    popupImage.src = event.target.style.backgroundImage.slice(5,-2);
-                    toggleBigSizeImage();
-                }
-            }
-        }
-    });
-
-    // нажатие на кнопку +
-    const button = document.querySelector(".user-info__button");
-    button.addEventListener("click", toggleFormAdd);
-
-    // закрытие формы добавления нового элемента
-    const crossButton = document.querySelector("#add-card .popup__close");
-    crossButton.addEventListener("click", toggleFormAdd);
-
-    // сабмит формы добавления карточки.
-    document.forms.new.addEventListener("submit", submitFormAdd);
-
-
-    // нажатие на кнопку Edit
-    const buttonEdit = document.querySelector(".button.user-info__edit");
-    buttonEdit.addEventListener("click", openFormProfile);
-
-    // закрытие формы редактирования профиля
-    const crossButtonEdit = document.querySelector("#profile .popup__close");
-    crossButtonEdit.addEventListener("click", closeFormProfile);
-
-    // сабмит формы редактирования профиля
-    document.forms.profile.addEventListener("submit", submitFormProfile);
-
-    // закрытие закрытия попапа с большой картинкой
-    const crossButtonBigImage = document.querySelector("#big-size-image .popup__close");
-    crossButtonBigImage.addEventListener("click", toggleBigSizeImage);
-
-    //валидация редактирования профиля
-    document.forms.profile.elements.name.addEventListener("input", validateProfileForm);
-    document.forms.profile.elements.job.addEventListener("input", validateProfileForm);
-
-    //валидация добавления новой карточки
-    document.forms.new.elements.name.addEventListener("input", validateAddCardForm);
-    document.forms.new.elements.link.addEventListener("input", validateAddCardForm);
+  return cardList;
 }
 
-// 0 - пустая строка
-// 1 - ок
-// 2 - слишком длинная или короткая
-
-function validateLenghtStr(str, min, max) {
-    if(str.length===0)
-        return 0;
-    if(str.length >= min && str.length <= max)
-        return 1;
-    return 2;
+function initAddCard(cardList, openImagePopup) {
+  const addCardPopupElement = document.querySelector('#add-card');
+  const addCardPopup = new Popup(addCardPopupElement);
+  const addCardNameField = new Field(document.forms.new.elements.name, validateStrLength, 'error-card-name');
+  const addCardLinkField = new Field(document.forms.new.elements.link, validateURL, 'error-card-link');
+  const addCardForm = new Form(document.forms.new, {
+    name: addCardNameField,
+    link: addCardLinkField,
+  });
+  const addCardbutton = document.querySelector('.user-info__button');
```

Comment:
Можно лучше: исправьте опечатку в названии переменной
```suggestion
  const addCardButton = document.querySelector('.user-info__button');
```